src/index.ts

```diff
@@ -96,6 +105,8 @@ export interface ProjectOptions {
    * as extraDependencies in the build target.
    */
   extraDependencies?: string[];
+
+  build?: ProjectBuildOptions|ProjectBuildOptions[];
```

Comment:
I'd rather avoid these types of unions, which require a type check before either accessing as an array or an individual object. With the optionality, that's two separate checks. I think it'd be better as always an Array.